victim-loss-identifier/src/fetcher.ts

```diff
@@ -0,0 +1,263 @@
+  Network,
+  Alchemy,
+  GetFloorPriceResponse,
+  FloorPriceMarketplace,
+} from "alchemy-sdk";
+import { LRUCache } from "lru-cache";
+
+export default class DataFetcher {
+  provider: providers.Provider;
+  private apiKeys: apiKeys;
+  private alchemy: Alchemy;
+  private ethPriceCache: LRUCache<number, number>;
+  private floorPriceCache: LRUCache<number, number>;
+
+  constructor(provider: providers.Provider, apiKeys: apiKeys) {
+    this.apiKeys = apiKeys;
+    this.provider = provider;
+    this.alchemy = new Alchemy({
+      apiKey: this.apiKeys.victimLossKeys.alchemyApiKey,
+      network: Network.ETH_MAINNET,
+    });
+    this.ethPriceCache = new LRUCache<number, number>({
+      max: 1000,
+    });
+    this.floorPriceCache = new LRUCache<number, number>({
+      max: 10000,
+    });
+  }
+
+  getTransactionReceipt = async (txHash: string) => {
+    let receipt;
+    let tries = 0;
+    const maxTries = 3;
+
+    while (tries < maxTries) {
+      try {
+        receipt = await this.provider.getTransactionReceipt(txHash);
+        break; // exit the loop if successful
+      } catch (err) {
+        tries++;
+        if (tries === maxTries) {
+          throw err; // throw the error if maximum tries reached
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000)); // wait for 1 second before retrying
+      }
+    }
+    return receipt;
+  };
+
+  getTransaction = async (txHash: string) => {
+    let receipt;
+    let tries = 0;
+    const maxTries = 3;
+
+    while (tries < maxTries) {
+      try {
+        receipt = await this.provider.getTransaction(txHash);
+        break; // exit the loop if successful
+      } catch (err) {
+        tries++;
+        if (tries === maxTries) {
+          throw err; // throw the error if maximum tries reached
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000)); // wait for 1 second before retrying
+      }
+    }
+    return receipt;
+  };
+
+  private getNativeTokenPrice = async (
+    timestamp: number
+  ): Promise<number | undefined> => {
+    if (this.ethPriceCache.has(timestamp)) {
+      console.log(`Using cached ETH price for timestamp: ${timestamp}`);
+      return this.ethPriceCache.get(timestamp);
+    }
+
+    const maxTries = 3;
+    let tries = 0;
+
+    while (tries < maxTries) {
+      try {
+        const url = `https://coins.llama.fi/prices/historical/${timestamp}/coingecko:ethereum`;
+        const response = await fetch(url);
+        const data = (await response.json()) as coinData;
+        const price = data.coins["coingecko:ethereum"]?.price;
+
+        if (price == null) {
+          tries++; // Increment tries counter for null price
+          console.log(`Fetched price is null (attempt ${tries})`);
+        } else {
+          this.ethPriceCache.set(timestamp, price);
+          return price;
+        }
+      } catch (e) {
+        tries++;
+        console.log(`Error in fetching ETH price (attempt ${tries}):`, e);
```

Comment:
```suggestion
        console.log(`Error in fetching ETH price (attempt ${tries}): `, e);
```
Very small suggestion to add a space for legibility of the log

Reply:
Same thing for L151